### DISCUSSION TITLE
upload_to_testflight an app built for Catalyst

### DISCUSSION THREAD

**Discussion by thibauddavid** [fastlane/fastlane#19771] [Q&A]
Hi !

Is there any way for upload_to_testflight to work with Testflight for Mac ? I have a lane which builds a Catalyst app from an iOS project, resulting in a .app being generated, but `upload_to_testflight` step only seems to support either ipa / pkg option ?

Thanks !

**Comment by thibauddavid** [comment]
I did ! 
In fact I had a Gemfile.lock at the root of my project, which was preventing fastlane to update. As a result, I was "stuck" on a version which didn't support correctly catalyst/Mac upload.

Once updated using fastlane_update, I can directly use `upload_to_testflight` without any parameters and it works out of the box for my pkg. (Note that once updated, `build_mac_app` correctly exports a .pkg from a generated .app).

Here is my lane:

```
  lane :testflight_catalyst do |options|
    # This step builds a .app, then exports as .pkg
    build_mac_app(
      scheme: ENV["SCHEME"],
      workspace: ENV["WORKSPACE_PATH"],
      destination: 'platform=macOS,arch=x86_64,variant=Mac Catalyst',
      export_method: "app-store"
    )
    # With an up to date fastlane, this step uploads .pkg to store out of the box
    upload_to_testflight(
      skip_waiting_for_build_processing: false
    )
  end
```

If you run into certificates issue let me know (particularly, you'll need a Mac Installer certificate to sign your .pkg, additionally to your Mac AppStore certificate)

**Reply by Thompsons21** [reply]
For our app, we’re using a separate app id to upload beta builds to Testflight separately from our App Store app. The easiest way might be to create a specific build configuration for it in Xcode, and then a scheme to compile/archive it. (It might be a good idea to also change the bundle name and/or app icon, to distinguish both.) Once setup, it’s hardly any manual work to create new build.  [TellCulvers Survey](https://tellculvers.org/)
